haystack_experimental/components/converters/file_to_image.py

```diff
@@ -0,0 +1,111 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import mimetypes
+from pathlib import Path
+from typing import Any, Dict, List, Literal, Optional, Union
+
+from haystack import component, logging
+from haystack.components.converters.utils import get_bytestream_from_source, normalize_metadata
+from haystack.dataclasses import ByteStream
+
+from haystack_experimental.components.converters.image_utils import DETAIL_TO_IMAGE_SIZE, encode_image_to_base64
+from haystack_experimental.dataclasses.chat_message import ImageContent
+
+logger = logging.getLogger(__name__)
+
+
+_EMPTY_BYTE_STRING = b""
+
+
+@component
+class ImageFileToImageContent:
+    """
+    Converts image files to ImageContent objects.
+    """
+
+    def __init__(self, *, detail: Optional[Literal["auto", "high", "low"]] = None, downsize: bool = False):
+        """
+        Create the ImageFileToImageContent component.
+
+        :param detail: Optional detail level of the image (only supported by OpenAI). One of "auto", "high", or "low".
+        :param downsize: If True, resizes the image to fit within the specified dimensions while maintaining aspect
+            ratio. This reduces file size, memory usage, and processing time, which is beneficial when working with
+            models that have resolution constraints or when transmitting images to remote services.
+        """
+        self.detail = detail
+        self.downsize = downsize
```

Comment:
@anakin87 do you think it would be worth to add a `pillow_import.check()` here if `detail` and `downsize` are provided? That's the condition for needing Pillow installed. 

The only downside I see is if the user overwrites these values at runtime such that Pillow is not needed anymore (e.g. downsize=False).

Reply:
ok, what about adding a similar check in the `run` method?

Reply:
added